platform/linux-generic/include/odp_parse_internal.h

```diff
@@ -76,39 +76,9 @@ int _odp_packet_parse_common_l3_l4(packet_parser_t *prs,
  * Returns 0 on success, 1 on packet errors, and -1 if the packet should be
  * dropped.
  */
-static inline int _odp_packet_parse_common(odp_packet_hdr_t *pkt_hdr,
```

Comment:
The idea is that we need only one non-inlined call (_odp_packet_parse_common) instead of two (_odp_parse_eth and _odp_packet_parse_common_l3_l4). I probably should have declared _odp_parse_eth and _odp_packet_parse_common_l3_l4 as inline. Can you try that?

Reply:
That doesn't seem to be that trivial since `_odp_packet_parse_common_l3_l4()` uses most of the inline functions in `odp_parse.c`. If `odp_packet_parse()` implementation wouldn't also use `_odp_parse_eth()` and `_odp_packet_parse_common_l3_l4()` this would be rather simple.

Reply:
Please try v3.

Reply:
Throughput seems better now, but there is still consistent overhead and some new pedantic build errors.

![Screen Shot 2022-05-31 at 17 12 03](https://user-images.githubusercontent.com/1427501/171194541-c0013306-2cd0-4b7c-9fbd-910b9d7b74aa.png)

Reply:
Ok, one more. Please try v4. If it doesn't perform, I'll drop the commit.